modules/extensions/galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3DynamicStatusStore.java

```diff
@@ -305,31 +305,66 @@ public String get(@NotNull String key) throws DynamicStatusStoreException {
     @Override
     public @NotNull Map<String, String> getPrefix(@NotNull String keyPrefix) throws DynamicStatusStoreException {
 
-        logger.debug("Etcd extension getting property with a prefix of "+keyPrefix);
+        Map<String, String> keyValues = new HashMap<>();
+
+        List<KeyValue> kvs = getPropertiesFromETCDWithPrefix(keyPrefix, false);
+
+        if (kvs.isEmpty()) {
+            return new HashMap<>();
+        }
+
+        for (KeyValue kv : kvs) {
+            keyValues.put(kv.getKey().toString(UTF_8), kv.getValue().toString(UTF_8));
+        }
+        return keyValues;
+    }
+
+    /**
+     * A get of all keys that start with a specified prefix. They are returned in a list.
+     * 
+     * @param keyPrefix - the prefix for any key(s)
+     * @return A list of keys
+     * @throws DynamicStatusStoreException A failure occurred.
+     */
+    @Override
+    public List<String> getPrefixKeysOnly(@NotNull String keyPrefix) throws DynamicStatusStoreException {
```

Comment:
Should this be returning a collection<String> rather than a list ? depends if there is some sorted order I guess.